tests/test_qasm_examples.py

```diff
@@ -1,17 +1,136 @@
 import os
 import re
+import sympy
 import pytest
+import numpy as np
 
 from qdata import parser
 
+
+def split_at_semicol(qasm_lines):
+    """Helper function to split lines at semicolons"""
+    form_lines = []
+    for line_1 in qasm_lines:
+        split_lines = line_1.split(";")
+        if len(split_lines) == 1:
+            form_lines.append(split_lines[0])
+        elif len(split_lines) == 2 and split_lines[1] == "":
+            form_lines.append(split_lines[0] + ";")
+        else:
+            for line_2 in split_lines:
+                delim = ";" * (clean(line_2) not in ["{", "}"])
+                if clean(line_2):
+                    form_lines.append(clean(line_2) + delim)
+    return form_lines
```

Comment:
I think this helper function can be improved by commenting each if branch, to explain what we are branching at, and the required fix